sql/rowexec/ddl_iters.go

```diff
@@ -131,55 +131,81 @@ func (l *loadDataIter) parseLinePrefix(line string) string {
 }
 
 func (l *loadDataIter) parseFields(ctx *sql.Context, line string) ([]sql.Expression, error) {
-	// Step 1. Start by Searching for prefix if there is one
+	// Start by searching for prefix if there is one
 	line = l.parseLinePrefix(line)
 	if line == "" {
 		return nil, nil
 	}
 
-	// Step 2: Split the lines into fields given the delim
-	fields := strings.Split(line, l.fieldsTerminatedBy)
-
-	// Step 3: Go through each field and see if it was enclosed by something
-	// TODO: Support the OPTIONALLY parameter.
+	// line is parsed character-by-character to respect enclosed fields that may contain the field terminator
+	var fields []string
+	var currentField strings.Builder
+	var encChar, escChar byte
 	if l.fieldsEnclosedBy != "" {
-		for i, field := range fields {
-			if field[0] == l.fieldsEnclosedBy[0] && field[len(field)-1] == l.fieldsEnclosedBy[0] {
-				fields[i] = field[1 : len(field)-1]
-			} else {
-				return nil, fmt.Errorf("error: field not properly enclosed")
-			}
-		}
+		encChar = l.fieldsEnclosedBy[0]
 	}
-
-	// Step 4: Handle the ESCAPED BY parameter.
 	if l.fieldsEscapedBy != "" {
```

Comment:
When this is empty string, we are now checking if certain characters are escaped by `NUL` character, which can cause problems.